File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	var sum = 0
	for i := 1; i <= n; i++ {
		sum += i
	}
	return sum * sum
}

func SumOfSquares(n int) int {
	var sumOfSquares = 0
	for i := 1; i <= n; i++ {
		sumOfSquares += i * i
	}
	return sumOfSquares
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
square_of_sum(n::Int) = sum(1:n)^2

"Sum the squares of the first `n` positive integers"
sum_of_squares(n::Int) = sum((1:n).^2)

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
difference(n::Int) = square_of_sum(n) - sum_of_squares(n)
